app/javascript/dashboard/components/widgets/conversation/ChatFilter.vue

```diff
@@ -1,27 +1,75 @@
 <template>
-  <select v-model="activeStatus" class="status--filter" @change="onTabChange()">
-    <option
-      v-for="(value, status) in $t('CHAT_LIST.CHAT_STATUS_FILTER_ITEMS')"
-      :key="status"
-      :value="status"
+  <div>
+    <woot-button
+      variant="smooth"
+      size="small"
+      color-scheme="secondary"
+      class="selector-button"
+      @click="toggleDropdown"
     >
-      {{ value['TEXT'] }}
-    </option>
-    <option value="all">
-      {{ $t('CHAT_LIST.FILTER_ALL') }}
-    </option>
-  </select>
+      {{ $t('CHAT_LIST.CHAT_STATUS_FILTER_ITEMS')[activeStatus]['TEXT'] }}
+      <fluent-icon
+        :icon="showActionsDropdown ? 'chevron-up' : 'chevron-down'"
+        class="icon"
+        size="14"
+      />
+    </woot-button>
+    <div
+      v-if="showActionsDropdown"
+      v-on-clickaway="closeDropdown"
+      class="dropdown-pane dropdown-pane--open"
+    >
+      <woot-dropdown-menu>
```

Comment:
The filter implementation in this file is very similar to the one in `AsigneeTypeFilter` and `SortByFilter`, except for the specific implementation of how the filter item looks. I am assuming we will have multiple of these across the UI. We can abstract this out into separate components

```vue
<WootDropDownFilter :title="titleToDisplay">
  <WootFilterItem 
    v-for="opt in options"
    value="opt"
    selected="opt === selectedFilter"
    @onSelect="setTab()"
  >
    {{ opt }}
  </WootFilterItem>
</WootDropDownFilter>
```

This will reduce the code footprint in their parent by a lot. It will ensure all the styles in the filters remain consistent in terms of UI and will be easy to test too 

<details>

<summary>WootDropDownFilter</summary>

```vue
<div v-on-clickaway="closeDropdown" class="dropdown-pane dropdown-pane--open">
  <woot-dropdown-menu>
    <woot-dropdown-sub-menu :title="title">
      <slot/>
    </woot-dropdown-sub-menu>
  </woot-dropdown-menu>
</div>
```

</details>

<details>
   <summary>WootFilterItem</summary>

```vue
<woot-dropdown-item>
  <woot-button
    variant="clear"
    color-scheme="secondary"
    size="small"
    class="filter-items"
    :class="{ active: selected }"
    @click="$emit('click', value)"
  >
    <slot/>
    <fluent-icon
      v-if="selected"
      icon="checkmark"
      size="14"
      class="icon"
    />
  </woot-button>
</woot-dropdown-item>
```
</details>





> P.S. these examples don't include the button that triggers the dropdown, but we can definitely include those in the `WootDropDownFilter` implementation. 

> P.P.S. We can also use v-model within the `WootDropDownFilter` component, which will involve going through the elements in its slots and altering their local state, for now, that seems unnecessary.

@iamsivin WDYT? 

CC: @pranavrajs @fayazara need thoughts

Reply:
@scmmishra Done can you take a look? https://github.com/chatwoot/chatwoot/pull/6186/commits/4b33ae8dca11f1574d222116169a94b60816d02d

Reply:
Looks good, resolving the comment